examples/mlperf/dataloader.py

```diff
@@ -139,6 +140,37 @@ def receive_batch():
     shm.close()
     shm.unlink()
 
+def load_bert_file(fn:str) -> List[dict]:
+  with open(fn, "rb") as f: data = pickle.load(f)
+  return data
+
+def process_batch_bert(data: List[dict]) -> dict[str, Tensor]:
+  return {
+    "input_ids": Tensor(np.concatenate([s["input_ids"] for s in data], axis=0), dtype=dtypes.float32),
+    "input_mask": Tensor(np.concatenate([s["input_mask"] for s in data], axis=0), dtype=dtypes.float32),
+    "segment_ids": Tensor(np.concatenate([s["segment_ids"] for s in data], axis=0), dtype=dtypes.float32),
+    "masked_lm_positions": Tensor(np.concatenate([s["masked_lm_positions"] for s in data], axis=0), dtype=dtypes.float32),
+    "masked_lm_ids": Tensor(np.concatenate([s["masked_lm_ids"] for s in data], axis=0), dtype=dtypes.float32),
+    "masked_lm_weights": Tensor(np.concatenate([s["masked_lm_weights"] for s in data], axis=0), dtype=dtypes.float32),
+    "next_sentence_labels": Tensor(np.concatenate([s["next_sentence_labels"] for s in data], axis=0), dtype=dtypes.float32),
+  }
+
+# For train: Stop when we run through all data
+# For val: Wrap around val dataset and never stop
```

Comment:
how do you plan to solve the issue if val is not a multiple of BS? for mlperf it needs to eval on data exactly once

Reply:
In this implementation, the model processes each data sample exactly once, regardless of the batch size. To my knowledge, in MLPerf, the validation dataset also wraps around after it has been fully processed.

Reply:
i see, can you add the link to this wrapping implementation in reference in comment?